### PR TITLE
[rocthrust] Successful build in clean chroot

### PR DESCRIPTION
**Pull request by tpkessler**
Reduce dependencies, tidy up build() and package() routines.

### CODE REVIEW

File: rocthrust/PKGBUILD [hunk 1]
```diff
@@ -1,50 +1,36 @@
...
+pkgrel=4
+pkgdesc="Port of the Thrust parallel algorithm library atop HIP/ROCm"
 arch=('x86_64')
 url="https://github.com/ROCmSoftwarePlatform/rocThrust"
-license=('custom:NCSAOSL')
-depends=('hcc' 'hip')
-makedepends=('cmake' 'hcc' 'python2' 'rocminfo' 'comgr')
-source=("https://github.com/ROCmSoftwarePlatform/rocThrust/archive/rocm-$pkgver.tar.gz")
+license=('Apache-2.0')
+depends=('hip-hcc' 'rocprim')
```
Comment (acxz) [inline rocthrust/PKGBUILD:9]: Shouldn't this be hip instead of the specific hip-hcc?

File: rocthrust/PKGBUILD [hunk 2]
```diff
@@ -1,50 +1,36 @@
 # Maintainer: Markus Näther <naetherm@informatik.uni-freiburg.de>
 pkgname=rocthrust
 pkgver=3.3.0
-pkgrel=3
-pkgdesc="Port of the Thrust parallel algorithm library atop HIP/ROCm."
+pkgrel=4
+pkgdesc="Port of the Thrust parallel algorithm library atop HIP/ROCm"
 arch=('x86_64')
 url="https://github.com/ROCmSoftwarePlatform/rocThrust"
```
Comment (acxz) [inline rocthrust/PKGBUILD:7]: Can we change the url to https://rocm-documentation.readthedocs.io/en/latest/ROCm_Libraries/ROCm_Libraries.html#rocthrust

File: rocthrust/PKGBUILD [hunk 3]
```diff
@@ -1,50 +1,36 @@
...
-license=('custom:NCSAOSL')
-depends=('hcc' 'hip')
-makedepends=('cmake' 'hcc' 'python2' 'rocminfo' 'comgr')
-source=("https://github.com/ROCmSoftwarePlatform/rocThrust/archive/rocm-$pkgver.tar.gz")
+license=('Apache-2.0')
+depends=('hip-hcc' 'rocprim')
+makedepends=('cmake' 'git' 'hcc')
+source=("$url/archive/rocm-$pkgver.tar.gz")
```
Comment (acxz) [inline rocthrust/PKGBUILD:11]: For the source can we add `${pkgname}-${pkgver}`? Like in the warning box here: https://wiki.archlinux.org/index.php/PKGBUILD#source

File: rocthrust/PKGBUILD [hunk 4]
```diff
@@ -1,50 +1,36 @@
...
   cd "$srcdir/build"
 
   make DESTDIR="$pkgdir" install
 
-  install -d "$pkgdir/etc/ld.so.conf.d"
-  cat << EOF > "$pkgdir/etc/ld.so.conf.d/rocthrust.conf"
+  install -Dm644 /dev/stdin "$pkgdir/etc/ld.so.conf.d/rocthrust.conf" << EOF 
 /opt/rocm/rocthrust/lib
 EOF
+  install -Dm644 "$srcdir/rocThrust-rocm-$pkgver/LICENSE" "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
```
Comment (acxz) [inline rocthrust/PKGBUILD:33]: The Apache-2.0 license is included in the `licenses` package and as such does not need to be installed here. See: https://wiki.archlinux.org/index.php/PKGBUILD#license

File: rocthrust/PKGBUILD (hunk 1 above)
Comment (tpkessler) [reply rocthrust/PKGBUILD:9]: rocthrust is for ROCm only. It's a port of the thrust library from CUDA.

File: rocthrust/PKGBUILD (hunk 2 above)
Comment (tpkessler) [reply rocthrust/PKGBUILD:7]: Sure. But shouldn't we change this for all packages in this repo then? At the moment they all point to the github repos. As far as I can see, the documentation of ROCm is just the README.md from the repos.

File: rocthrust/PKGBUILD (hunk 3 above)
Comment (tpkessler) [reply rocthrust/PKGBUILD:11]: Good point. We should change this for all packages.

File: rocthrust/PKGBUILD (hunk 4 above)
Comment (tpkessler) [reply rocthrust/PKGBUILD:33]: Thanks, I didn't know that. I've removed the line (and also the version number as apache has version 2 in the license package)

File: rocthrust/PKGBUILD (hunk 1 above)
Comment (acxz) [reply rocthrust/PKGBUILD:9]: I see kk

File: rocthrust/PKGBUILD (hunk 2 above)
Comment (acxz) [reply rocthrust/PKGBUILD:7]: We should but for now lets just change them one at a time as they come up since it is not a critical issue.

Comment (tpkessler) [reply rocthrust/PKGBUILD]: I've changed the url.

**Pull request merged**